### ISSUE TITLE
False positive `deprecated-class` when the class is imported in a guarded import block

### ISSUE THREAD

**Issue by perrinjerome** [pylint-dev/pylint#9533]
### Current problem

```py
import sys

if sys.version_info > (2, ):
    from collections.abc import Set
else:
    from collections import Set

isinstance(set(), Set)
```

```console
$ pylint --disable=C a.py
************* Module a
a.py:6:4: W4904: Using deprecated class Set of module collections (deprecated-class)

-----------------------------------
Your code has been rated at 8.00/10
```

### Desired solution

In https://github.com/pylint-dev/pylint/pull/4702 pylint was extended with a concept of "guarded import blocks" and it does not trigger `import-error` and similar errosr when the import is made in a guarded import block.

I believe it would be nice to have same behavior for `deprecated-class` as well.

**Comment by SaharYaz** [comment]
Hi! I'd like to work on this issue. I reproduced the `deprecated-class` false positive against the current `main` branch.

My initial plan is to make `deprecated-class` skip `from ... import ...` checks when the import is directly inside a recognized `sys.version_info` guard, while keeping the warning for ordinary unguarded deprecated imports. I'll add focused regression tests for both guarded and unguarded cases.

Please let me know if that scope sounds appropriate or if you prefer a different approach.

**Comment by Pierre-Sassoulas** [comment]
Hey @SaharYaz, sounds good but search for a pre-existing function to detect sys guard, don't reimplement it :)
